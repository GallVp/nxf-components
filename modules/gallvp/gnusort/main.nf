process GNUSORT {
    tag "$meta.id"
    label 'process_low'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/ubuntu:20.04':
        'nf-core/ubuntu:20.04' }"

    input:
    tuple val(meta), path(input_file)

    output:
    tuple val(meta), path("*.${extension}")                   , emit: sorted
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    extension = input_file.extension
    """
    sort --parallel=${task.cpus} \\
        $args \\
        $txt_file \\
        > ${prefix}.${extension}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sort: \$(sort --version | sed -n '/sort (GNU coreutils) / s/sort (GNU coreutils) //p')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    
    touch ${prefix}_sorted.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sort: \$(sort --version | sed -n '/sort (GNU coreutils) / s/sort (GNU coreutils) //p')
    END_VERSIONS
    """
}
