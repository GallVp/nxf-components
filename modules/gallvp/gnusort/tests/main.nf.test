nextflow_process {

    name "Test Process GNUSORT"
    script "../main.nf"
    process "GNUSORT"

    tag "modules"
    tag "modules_gallvp"
    tag "gnusort"

    test("nanopore - test_sequencing_summary") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/nanopore/sequencing_summary/test.sequencing_summary.txt', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }
    test("nanopore - test_sequencing_summary stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }

}
