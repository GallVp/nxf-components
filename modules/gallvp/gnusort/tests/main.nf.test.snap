{
    "nanopore - test_sequencing_summary stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_sorted.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,72606742313f58d65d9582367549c581"
                ],
                "sorted_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test_sorted.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,72606742313f58d65d9582367549c581"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-03-18T23:21:05.927274941"
    },
    "nanopore - test_sequencing_summary": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_sorted.txt:md5,56ef18b392e17b5f1dd9309f7983c867"
                    ]
                ],
                "1": [
                    "versions.yml:md5,72606742313f58d65d9582367549c581"
                ],
                "sorted_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test_sorted.txt:md5,56ef18b392e17b5f1dd9309f7983c867"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,72606742313f58d65d9582367549c581"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-03-18T23:20:57.965241975"
    }
}